learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,240 @@
+
+**Search rules** must be a JSON object specifying the restrictions that will be applied to search requests on a given index. It must contain at least one search rule. [To learn more about search rules, take a look at our tenant token payload reference.](#search-rules)
+
+As its name indicates, **API key** must be a valid Meilisearch API key. A tenant token will have access to the same indexes as the API key used when generating it. If no API key is provided, the SDK might be able to infer it automatically.
+
+**Expiration date** is optional when using an SDK. Tokens become invalid after their expiration date. Tokens without an expiration date will expire when their parent API key does.
+
+You can read more about each element of a tenant token payload in [this guide's final section](#tenant-token-payload-reference).
+
+### Using a tenant token with an SDK
+
+After creating a token, you can sent it back to the front-end. There, you can use it to make queries that will only return results whose `user_id` attribute equals the current user's ID:
+
+<CodeSamples id="tenant_token_guide_search_sdk_1" />
+
+Applications may use tenant tokens and API keys interchangeably when searching. For example, the same application might use a default search API key for queries on public indexes and a tenant token for logged-in users searching on private data.
+
+## Generating tenant tokens without a Meilisearch SDK
+
+Though Meilisearch recommends [using an official SDK to generate tenant tokens](#generating-tenant-tokens-with-an-sdk), this is not a requirement. Since tenant tokens follow the [JWT standard](https://jwt.io), you can use a number of [compatible third-party libraries](https://jwt.io/libraries). You may also skip all assistance and generate a token from scratch, though this is probably unnecessary in most production environments.
+
+If you are already familiar with the creation of JWTs and only want to know about the specific requirements of a tenant token payload, skip this section and take a look at the [token payload reference](#tenant-token-payload-reference).
+
+### Generating a tenant token with a third-party library
+
+Using a third-party library for tenant token generation is fairly similar to creating tokens with an official SDK. The following example uses the [`node-jsonwebtoken`](https://github.com/auth0/node-jsonwebtoken) library:
+
+```js
+const jwt = require('jsonwebtoken');
+
+const apiKey = 'my_api_key';
+const currentUserID = 'a_user_id';
+
+const tokenPayload = {
+  searchRules: {
+    'patient_medical_records': {
+      'filter': `user_id = ${currentUserID}`
+     }
+  },
+  apiKeyPrefix: apiKey.substring(0, 8),
+  exp: parseInt(Date.now() / 1000) + 20 * 60 // 20 minutes
+};
+
+const token = jwt.sign(tokenPayload, apiKey, {algorithm: 'HS256'});
```

Comment:
Very cool example 👌. If you want to shorten it, by default `jsonwebtoken` uses `HS256` as default algorithm :) But it is oke to leave it for clarity if you prefer

Reply:
Yeah, I was about to do that, based on the samples you sent over slack, but I realised it was useful to keep it. That option gives me a good hook to list the encryption algorithms Meilisearch supports.

Reply:
I agree !